### ISSUE TITLE
Style/RedundantParentheses multiline list detection problem

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#10883]
Consider

```ruby
[
  (
    1
  ),
  2
]
```

`Style/RedundantParentheses` doesn't detect a problem in this example. If we take out the second list element, the cop triggers:

```ruby
[
  (
    1
  )
]
```

```
[Correctable] Style/RedundantParentheses: Don't use parentheses around a literal
```

## Expected behavior

Both examples should trigger `Style/RedundantParentheses`

## RuboCop version

```
1.33.0 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 2.6.9 x86_64-darwin21)
  - rubocop-minitest 0.21.0
  - rubocop-performance 1.14.3
  - rubocop-rails 2.15.2
```

**Comment by dvandersluis** [comment]
This was added purposely in #2874, possibly due to a parser issue.

**Comment by dvandersluis** [comment]
We might be able to handle this now, I'm looking at it.

**Comment by dvandersluis** [comment]
Hi @vlad-pisanov, your two issues are both handled in my PR now. Thanks for the report!
